### DISCUSSION TITLE
Max Line Width and Max Line Count Commands

### DISCUSSION THREAD

**Discussion by Joelson-Forte** [Purfview/whisper-standalone-win#87] [Ideas]
Please, if possible add the following commands:

--max_line_width MAX_LINE_WIDTH
(requires --word_timestamps True) the maximum number of characters in a line before breaking the line (default: None)

--max_line_count MAX_LINE_COUNT
(requires --word_timestamps True) the maximum number of lines in a segment (default: None)

These commands are very useful and are greatly missed.

Thanks!

**Comment by Purfview** [comment]
Dupe of [Add support for --max_line_width --max_line_count](https://github.com/Purfview/whisper-standalone-win/issues/18).

**Comment by Joelson-Forte** [comment]
I undestand. Your job is very good. 
Thank you for all your effort!

**Comment by Purfview** [comment]
If OpenAI patch those options and makes them equal or better than what you can get from Subtitle Edit then I'll add them.
Now those commands are pointless as they produce borked subtitles.

**Reply by Joelson-Forte** [reply]
I already use Subtitle Edit to adjust common subtitles with one or two lines. The result is satisfactory, but the biggest problem is that the division of Subltitle Edit disfigures the original timecode of the Sentence.

And to make this worse, one of the options I like most in Whisper is the --highlight_words HIGHLIGHT_WORDS command, which highlights word by word.

As you know, with this option the subtitles are duplicated word for word until they form the complete sentence. Using the Subtitle Editin this case is impossible, because Subtitle Edit in this specific case, disfigures the entire subtitle.

Fortunately, Aegisub does a great job and adjusts the subtitles as best as possible so that they do not exceed the Horizontal 16:9 and Vertical 9:16 screen limit. But there are still a lot of words when the screen direction is vertical 9:16.

Hopefully this will be fixed someday. Until then I will eagerly wait.

Thank you very much!

**Reply by Purfview** [reply]
> I already use Subtitle Edit to adjust common subtitles with one or two lines. The result is satisfactory, but the biggest problem is that the division of Subltitle Edit disfigures the original timecode of the Sentence.

No, SE doesn't touch timestamps, unless you chose to correct them.
Actually that is what --max_line_width and --max_line_count does.

Here is example, from left to right - 1) No breaks, 2) --max_line_width=48 --max_line_count=2, 3) No breaks + SE's line breaks:

![alt text](https://i.imgur.com/rjCSCtT.png)

**Reply by Purfview** [reply]
>And to make this worse, one of the options I like most in Whisper is the --highlight_words HIGHLIGHT_WORDS command, which highlights word by word.

> As you know, with this option the subtitles are duplicated word for word until they form the complete sentence. Using the Subtitle Editin this case is impossible, because Subtitle Edit in this specific case, disfigures the entire subtitle.

Same for this, I don't see "disfigures" effect in SE, when I see borked subs output with max_line_width --max_line_count and some lines are not highlighted.

**Reply by Joelson-Forte** [reply]
See my screen recording to better understand what I'm trying to say. Sorry for the bad English, I use Google Translate and the words don't always correspond to what I really want to express.

https://drive.google.com/file/d/1QyLNCVGOfDKRXMdvIRck3CyT7SZbIJBp/view?usp=sharing

In this video I transcribed it using the  --highlight_words HIGHLIGHT_WORDS command. The generated srt file is single line and some sentences are very long. Although the video player does a great job of automatically splitting the subtitle into two lines, when the srt files are imported into an editing program, the problem of long lines is noticed more clearly. This is also easy to fix in the editing program, just split the long line into two lines and the problem is solved.

Subtitle Edit is another good solution is to use to break the lines. However, when the --highlight_words HIGHLIGHT_WORDS command is used in Whisper and you use Subtitle Edit to break the lines, at least here on my system, what you see in my screen recording occurs. And for that reason alone I suggested the --max_line_width MAX_LINE_WIDTH and --max_line_count MAX_LINE_COUNT options in the hope that these options would fix this.

**Reply by Purfview** [reply]
After watching your video I see what you meant.

**Comment by Purfview** [comment]
After OP's comment quoted below, I've changed this idea's status "maybe" to "invalid". 😄

> I was testing Whisper --max_line_width MAX_LINE_WIDTH and --max_line_count MAX_LINE_COUNT commands and saw that the result wasn't good, impossible to use. You did well to disable this in Whisper-Faster.

Read more in [Output every word in a separate line](https://github.com/Purfview/whisper-standalone-win/discussions/107).

**Reply by Joelson-Forte** [reply]
In any case, I will continue testing with the new versions that are released until one day this is improved in Whisper Open AI and becomes valid. 😄

**Comment by Purfview** [comment]
There is alternative -> [New args: sentence, standard, max_line_width, max_line_count, max_comma, max_gap.](https://github.com/Purfview/whisper-standalone-win/discussions/125)
In r189.1 `--highlight_words` supports `--max_line_width` and `--max_line_count`.
